Description: destroy socket on DNS error
 The socket was never destroyed on DNS errors. This broke some clients, including
 lib/https.js.
Origin: upstream, https://github.com/joyent/node/commit/4671e54
---
 lib/net.js |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/lib/net.js b/lib/net.js
index 8547941..bdd5ba8 100644
--- a/lib/net.js
+++ b/lib/net.js
@@ -570,6 +570,7 @@ Socket.prototype.connect = function(port /* [host], [cb] */) {
         // error event to the next tick.
         process.nextTick(function() {
           self.emit('error', err);
+          self.destroy();
         });
       } else {
         timers.active(self);
-- 
1.7.5.4

